src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -1201,6 +1201,33 @@ public function getInvalidBinaryData()
             ),
         );
     }
+
+    public function testParsePhpConstant()
+    {
+        $yaml = '!php/const:PHP_INT_MAX';
+
+        // Returns a string by default
+        $this->assertSame($yaml, $this->parser->parse($yaml));
+        // Returns a constant value when passing flag
+        $this->assertSame(PHP_INT_MAX, $this->parser->parse($yaml, Yaml::PARSE_CONSTANT));
+    }
+
+    public function testParseNestedPhpConstant()
+    {
+        $yaml = "foo: '!php/const:PHP_INT_MAX'";
```

Comment:
The tag will never be evaluated as you quoted the value (this is then just a simple string in YAML).